perf-tool/src/monitor.cpp

```diff
@@ -36,53 +36,8 @@ using json = nlohmann::json;
 std::atomic<int> monitorSampleCount{0};
 EventConfig monitorConfig;
 
-JNIEXPORT void JNICALL MonitorContendedEntered(jvmtiEnv *jvmtiEnv, JNIEnv *env, jthread thread, jobject object){
-    json j;
-    jvmtiError error;
-
-    /* Get number of events and increment */
-    int numSamples = atomic_fetch_add(&monitorSampleCount, 1);
-    if (numSamples % monitorConfig.getSampleRate() != 0)
-        return;
-
-    jvmtiError err;
-    jint hash;
-    err = jvmtiEnv->GetObjectHashCode(object, &hash);
-    if (check_jvmti_error(jvmtiEnv, err, "Unable to retrieve object hashcode.")) {
-	    char str[32];
-	    sprintf(str, "0x%x", hash);
-	    j["monitorHash"] = str;
-    }
-
-    static std::map<std::string, int> numContentions;
-    jclass cls = env->GetObjectClass(object);
-    /* First get the class object */
-    jmethodID mid = env->GetMethodID(cls, "getClass", "()Ljava/lang/Class;");
-    jobject clsObj = env->CallObjectMethod(object, mid);
-    /* Now get the class object's class descriptor */
-    cls = env->GetObjectClass(clsObj);
-    /* Find the getName() method on the class object */
-    mid = env->GetMethodID(cls, "getName", "()Ljava/lang/String;");
-    /* Call the getName() to get a jstring object back */ 
-    jstring strObj = (jstring)env->CallObjectMethod(clsObj, mid);
-    /* Now get the c string from the java jstring object */
-    const char *str = env->GetStringUTFChars(strObj, NULL);
-    /* record calling class */
-    j["Class"] = str;
-    
-    /* free str */
-   
-    static std::mutex contentionMutex;
-    int samplesPerObjectType = 0;
-    {
-    std::lock_guard<std::mutex> lg(contentionMutex);
-    samplesPerObjectType = ++(numContentions[std::string(str)]);
-    }
-    j["numTypeContentions"] = samplesPerObjectType;
-    /* Release the memory pinned char array */
-    env->ReleaseStringUTFChars(strObj, str);
-    env->DeleteLocalRef(cls);
-
+void getStackTrace(jvmtiEnv *jvmtiEnv, jthread thread, json& j)
+{
     if (monitorConfig.getStackTraceDepth() > 0)
```

Comment:
We would like to use this `getStackTrace()` method in other routines as well, not just for monitors.
To make it more generic, instead of using `monitorConfig.getStackTraceDepth()`, we need to pass a parameter with the desired stack trace depth. Also, this new method probably should stay in infra.cpp.